OpenSim/Common/AbstractDataTable.cpp

```diff
@@ -132,28 +132,21 @@ AbstractDataTable::setColumnLabel(const size_t columnIndex,
     OPENSIM_THROW_IF(!hasColumnLabels(),
                      NoColumnLabels);
 
-    ValueArray<std::string> newLabels{};
-    const auto& oldLabels = 
-        _dependentsMetaData.getValueArrayForKey("labels");
+    auto labels = getColumnLabels();
 
-    OPENSIM_THROW_IF(columnIndex >= oldLabels.size(),
-                     ColumnIndexOutOfRange,
-                     columnIndex, 0, 
-                     static_cast<unsigned>(oldLabels.size() - 1));
-
-    for(unsigned i = 0; i < oldLabels.size(); ++i) {
-        if(i == columnIndex) {
-            newLabels.upd().push_back(Value<string>(columnLabel));
-        } else {
-            auto value = Value<string>(oldLabels[i].getValue<string>());
-            newLabels.upd().push_back(value);
-        }
+    if(columnIndex == labels.size()) {
```

Comment:
I think this is a change in the behavior. Doesn't this allow having more column labels than there are columns?

Reply:
Funny, I thought that was it was doing previously so went out of my way to add it.
I was clearly wrong since the previous `newLabels.upd().push_back()`  was limit to i < labels.size(). I'll revert this.